src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContent.cs

```diff
@@ -35,14 +33,24 @@ public abstract class TagHelperContent : IEnumerable<string>, IHtmlContent
         /// </summary>
         /// <param name="value">The <see cref="string"/> that replaces the content.</param>
         /// <returns>A reference to this instance after the set operation has completed.</returns>
-        public abstract TagHelperContent SetContent(string value);
+        public virtual TagHelperContent SetContent(string value)
+        {
+            Clear();
+            Append(value);
+            return this;
+        }
 
         /// <summary>
         /// Sets the content.
         /// </summary>
-        /// <param name="tagHelperContent">The <see cref="TagHelperContent"/> that replaces the content.</param>
+        /// <param name="htmlContent">The <see cref="IHtmlContent"/> that replaces the content.</param>
         /// <returns>A reference to this instance after the set operation has completed.</returns>
-        public abstract TagHelperContent SetContent(TagHelperContent tagHelperContent);
+        public virtual TagHelperContent SetContent(IHtmlContent htmlContent)
+        {
+            Clear();
+            Append(htmlContent);
+            return this;
+        }
```

Comment:
This class is `abstract` to clean up the user-visible interface, not for extensibility. Remove `virtual`.